From 2bdeb7b0e7dd100fce9e1f4d1ecf1cd6b5b9702c Mon Sep 17 00:00:00 2001
From: Matthew Craven <clyring@gmail.com>
Date: Tue, 8 Aug 2023 14:25:03 -0400
Subject: [PATCH] Use a more specific include than Rts.h (#607)

The Rts.h include seems to cause build problems.
See also:
  https://gitlab.haskell.org/ghc/ghc/-/issues/23789
  https://github.com/haskell/bytestring/issues/606

(cherry picked from commit e8a9b522063a203424eac3357873e4a8f2a09965)
---
 cbits/is-valid-utf8.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/libraries/bytestring/cbits/is-valid-utf8.c
===================================================================
--- a/libraries/bytestring/cbits/is-valid-utf8.c
+++ b/libraries/bytestring/cbits/is-valid-utf8.c
@@ -50,7 +50,7 @@ SUCH DAMAGE.
 #endif
 
 #include <MachDeps.h>
-#include "Rts.h"
+#include "ghcplatform.h"
 
 #ifdef WORDS_BIGENDIAN
 #define to_little_endian(x) __builtin_bswap64(x)
